tokio/tests/async_send_sync.rs

```diff
@@ -454,6 +454,7 @@ assert_value!(tokio::task::JoinSet<NN>: !Send & !Sync & Unpin);
 assert_value!(tokio::task::JoinSet<YN>: Send & Sync & Unpin);
 assert_value!(tokio::task::JoinSet<YY>: Send & Sync & Unpin);
 assert_value!(tokio::task::LocalSet: !Send & !Sync & Unpin);
+assert_value!(tokio::task::coop::RestoreOnPending: !Send & !Sync & Unpin);
```

Comment:
It probably should not be `UnwindSafe` either.

Reply:
@Darksonn I was trying to come up with a test for this following the examples I could find in the codebase. I tried adding the case below to see if it was currently `UnwindSafe`, but that fails to compile. I'm not sure how I can add a negative assertion for this.

```
fn restore_on_pending_is_unwind_safe() {
    is_unwind_safe::<tokio::task::coop::RestoreOnPending>();
}
```

Reply:
If a type is `UnwindSafe` that means you can return it through a `catch_unwind` boundary, but I doubt that's correct for the end-user. That said, `UnwindSafe` is more of a lint than the other traits, so it's not so important.

Regarding tests, we would need to add it to `async_send_sync.rs` similar to the existing three auto traits. But you don't have to do that in this PR.

Reply:
I opened #7413.

Reply:
Thanks for the explanation, that confirms how I understood it. From a user perspective it doesn't seem to make sense to let this type cross a catch_unwind indeed. Not sure if there's a correctness problem though.